server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -669,8 +671,97 @@ private void updateMergeState(Map<TableId,MergeStats> mergeStatsCache) {
     }
   }
 
+  private Text getDeletionStartRow(MergeInfo info) throws AccumuloException {
+    // Find the first tablet in the deletion range and check if it was fenced
+    // If the tablet has been fenced we need to keep it as there is still valid
+    // data and we don't want to delete it
+    Text deletionStartRow = info.getExtent().prevEndRow();
+    if (deletionStartRow != null) {
+      Manager.log.trace("Finding tablet that contains the start row {} for deletion",
+          deletionStartRow);
+      KeyExtent firstTablet =
+          loadTabletMetadata(info.getExtent().tableId(), deletionStartRow, ColumnType.PREV_ROW)
+              .map(TabletMetadata::getExtent).orElse(null);
+      // If needsFencingForDeletion() is true then this tablet was previously fenced
+      // so we need to keep it and not delete the tablet. Skip to the next highest
+      // tablet after this one if it exists
+      if (firstTablet != null && needsFencingForDeletion(info, firstTablet)) {
```

Comment:
You are right, this check is not needed, it was just force of habit of checking for null and I will get rid of it. I can also get rid of it for looking up the last tablet, something should always return since it's using the overlapping() method for looking up tablets so rows should always map to something.